library/ssl_tls.c

```diff
@@ -7652,10 +7652,14 @@ void mbedtls_ssl_config_free( mbedtls_ssl_config *conf )
     if( conf->psk != NULL )
```

Comment:
wouldn't it be better to change this line to 
`if( conf->psk != NULL || conf->psk_identity != NULL )`, as done in `mbedtls_ssl_conf_psk()`?

Reply:
I totally agree. Thank you for your review.

Reply:
@RonEld I disagree as we don't want to call free on one pointer while we got here due to other. I mean if we get inside this block due to ```conf->psk == NULL``` we don't want to do ```mbedtls_free( conf->psk_identity );```. Hence separation is good.